src/chips/AbstractChip.ts

```diff
@@ -0,0 +1,77 @@
+import {chips} from "../types/strategy/chips";
+import {HassServiceTarget} from "home-assistant-js-websocket";
+import {LovelaceChipConfig} from "../types/lovelace-mushroom/utils/lovelace/chip/types";
+import {Helper} from "../Helper";
+import {generic} from "../types/strategy/generic";
+
+/**
+ * Abstract Chip class.
+ *
+ * To create a new chip, extend this one.
+ *
+ * @class
+ * @abstract
+ */
+class AbstractChip {
+  /**
+   * Configuration of the chip.
+   *
+   * @type {chips.LovelaceChipOptions}
+   * @readonly
+   */
+  readonly #config: chips.LovelaceChipOptions;
+
+  /**
+   * The type of the chip.
+   *
+   * @type {LovelaceChipConfig["type"]}
+   * @private
+   * @readonly
+   */
+  readonly #type: LovelaceChipConfig["type"];
+
+  /**
+   * Class Constructor.
+   *
+   * @param {string} type The type of the chip.
+   * @param {chips.LovelaceChipOptions} options Chip options.
+   */
+  constructor(type: LovelaceChipConfig["type"], options: chips.LovelaceChipOptions = {}) {
+    this.#type = type;
+    this.#config = options;
+  }
+
+  /**
+   * Set the target to switch.
+   *
+   * @param {HassServiceTarget} target Target to switch.
+   */
+  setTarget(target: HassServiceTarget) {
+    if ("tap_action" in this.#config && generic.isCallServiceActionConfig(this.#config.tap_action)) {
+      this.#config.tap_action.target = target;
```

Comment:
This line `throws: TypeError: "target" is read-only` when the dashboard reloads.
E.g. by navigating from the Mushroom-Strategy Dashboard to another one and back.

@johanfrick I can't find the reason why this property is read-only, and why it's read-only on consecutive calls only. Would you mind to give it a try to resolve this issue?

Reply:
Configs in Lovelace area always immutable. Two common options are:

1. Make a deep copy to remove immutability in the constructor:
```javascript
    this.#config = JSON.parse(JSON.stringify(options));
```
OR

2. Remove readonly, then use spread operator to create a copy:
```javascript
      this.#config = {...{tap_action: {...this.#config.tap_action, target: target}}, ...this.#config};
```

Reply:
`this.#config = JSON.parse(JSON.stringify(options));`

Yeah, I've seen such suggestions, but it feels a bit like a "work-around".
I tried to figure out where the object becomes immutable, but no luck.

Reply:
Yeah, I don't think you'll find anything good in that rabbit hole.

Reply:
I just don't get it. The variables and values are the ones we define in the strategy. No issue there afaia. Then I thought it must be in the type/interface definitions, but I didn't find immutables there either.